include/xlList.h

```diff
@@ -0,0 +1,30 @@
+#ifndef _XL_LIST_H
+#define _XL_LIST_H
+
+#include "types.h"
+
+typedef struct list_item_s list_item_t;
+
+struct list_item_s{
+    list_item_t *next;
+    char data[];
```

Comment:
```suggestion
    u8 data[];
```
Does `u8` or `s8` work in place of all the `char` declarations/casts, both in this file and `xlObject.c`?